src/libNetwork/P2PComm.cpp

```diff
@@ -378,13 +391,29 @@ void P2PComm::ClearBroadcastHashAsync(const vector<unsigned char>& message_hash)
     m_broadcastToRemove.emplace_back(message_hash, chrono::system_clock::now());
 }
 
-void P2PComm::HandleAcceptedConnection(int cli_sock, Peer from)
+void P2PComm::EventCallback(struct bufferevent* bev, short events,
+                            [[gnu::unused]] void* ctx)
 {
-    //LOG_MARKER();
+    if (events & BEV_EVENT_ERROR)
+    {
+        LOG_GENERAL(WARNING, "Error from bufferevent");
+    }
 
-    LOG_GENERAL(INFO, "Incoming message from " << from);
+    // Not all bytes read out
+    if (!(events & (BEV_EVENT_EOF | BEV_EVENT_ERROR)))
+    {
+        return;
+    }
 
-    SocketCloser cli_sock_closer(&cli_sock, close_socket);
+    // Get the data stored in buffer
+    struct evbuffer* input = bufferevent_get_input(bev);
+    size_t len = evbuffer_get_length(input);
+    vector<unsigned char> message(len);
+    evbuffer_copyout(input, message.data(), len);
+    evbuffer_drain(input, len);
```

Comment:
error handling for the return value of `evbuffer_drain`

Reply:
Got it. Will add more error handling for the rest too.